src/audio/host-zephyr.c

```diff
@@ -489,10 +490,15 @@ static uint32_t host_get_copy_bytes_normal(struct comp_dev *dev)
 	else
 		avail_bytes = audio_stream_get_avail_bytes(&buffer_c->stream);
 
+	copy_bytes = MIN(avail_bytes, free_bytes);
```

Comment:
@makarukp Can you pease help me understand how this is supposed to work? If you want to copy as much data as possible from/to the DMA with fast-mode, don't you need to look at the direction of transfer and the type of copier as well? Take playback for example, assuming we have 100ms audio data available in the host DMA buffer, we would like to copy all of it, right? But for capture, we can possibly only copy what's available in the source buffer right?

Reply:
@ranj063 The transfer is always limited upwards by the size of internal buffer in copier. At this moment this limitation is equal to 2*ibs, so even if we have 100ms audio data available, there is limitation from free bytes which limits the transfer to 2 ms. 
Normally without FAST MODE copier is limited to 1 ms transfer. After enable it could maximize the transfer upt to 2ms at once.

Reply:
@ranj063 I don't think this FAST MODE is suitable for deep buffer, or at least it doesn't alone solve the problem. We need some additional logic to prevent host copier to transfer data unless certain minimum threshold in copier internal buffer is reached... e.g. if you copy 100ms of data in tick 1, you should not copy 1ms more in tick 2, but you should wait until copier internal buffer is almost empty, and then transfer another 100ms. This way we'd get some actual benefit from the buffer, right.

But seems this fast-mode doesn't really help and some other configuration of the copier is needed additionally to realize the benefits.

Reply:
@makarukp From the commit description it seems that this fast mode is only suitable for compressed data, for which the period definitions means nothing in the first place?

If that is the case, then clearly it's not suitable for PCM deep-buffer where we want to fetch large amounts from DDR but in 'predictable' ways linked to some sort of watermark/buffer-fullness.

You really need to clarify what this fast_mode is